autoload/phpactor/fzf.vim

```diff
@@ -0,0 +1,109 @@
+function! phpactor#fzf#quickfix(results) abort
+  let formated = s:align_pairs(keys(a:results), '^\(.\{-}:\d\+:\d\+:\)\s*\(.*\)\s*$', 100)
+  let tmp = copy(a:results)
+  let results = {}
+  for key in keys(tmp)
+    let newKey = formated[key]
+    let results[newKey] = tmp[key]
+  endfor
+  unlet tmp
+
+  function! s:quickfix(items) abort
+      call setqflist(a:items)
+      cw
+  endfunction
+
+  let actions = {
+        \ 'ctrl-t': 'tab split',
+        \ 'ctrl-x': 'split',
+        \ 'ctrl-v': 'vsplit',
+        \ 'ctrl-q': function('<SID>quickfix')
+        \ }
+
+  call fzf#run(fzf#wrap('find_references', fzf#vim#with_preview({
+        \ 'source': keys(results),
+        \ 'down': '60%',
+        \ '_action': actions,
+        \ 'sink*': function('<SID>quickfix_sink', [results, actions]),
+        \ 'options': [
+        \ '--expect='. join(keys(actions), ','),
+        \ '--multi',
+        \ '--bind=ctrl-a:select-all,ctrl-d:deselect-all',
+        \ '--inline-info',
+        \ '--header', ":: Press \x1b[35mCTRL-Q\x1b[m to open the quickfix with your selection",
+        \ '--delimiter=:', '--nth=1,4'
+        \ ]}, 'up', '?'), 1))
+endfunction
+
+function! s:align_pairs(list, regexp, ...) abort
+    let maxlen = 0
+    let pairs = {}
+    for elem in a:list
+        let match = matchlist(elem, a:regexp)
+        let [filename, text] = match[1:2]
+        let maxlen = max([maxlen, len(filename)])
+        let pairs[elem] = [filename, text]
+    endfor
+
+    let args = copy(a:000)
+    let max = 60
+    if 0 < len(args) && type(v:t_number) == type(args[0])
+        let max = remove(args, 0)
+    endif
+
+    let maxlen = min([maxlen, max])
+
+    return map(pairs, "printf('%-'.maxlen.'s', v:val[0]).' '.v:val[1]")
+endfunction
+
+function! s:quickfix_sink(results, actions, lines) abort
+  if 2 > len(a:lines)
+    return " Don't know how to handle this, should not append
```

Comment:
should this be an Exception ?

Reply:
I guess, I replicate most of the `s:common_sink` from fzf here so I didn't think too much about it.
Since it should not append until we messed up in the code I guess it could not hurt.

Reply:
I figure out why it was done this way while creating the exception.
The goal is to ensure that we got two lines at least which mean one "action" and at least one line selected.
But if you enter a filter that doesn't match with anything but still want to press enter to leave the window then we would go an exception.
Which is not what we want, so I think we should leave it like that.